### PR TITLE
b/140446546: add PrintOptions for transcoder

### PR DESCRIPTION
**Pull request by TAOXUY**
- Add flags to flip those print options
- Add an integration test

Here is the doc for those [options](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/transcoder/v2/transcoder.proto)

### CODE REVIEW

File: src/go/options/common.go [hunk 1]
```diff
@@ -77,26 +81,28 @@ type IAMCredentialsOptions struct {
...
+		TracingOutgoingContext:                "",
+		TracingMaxNumAttributes:               32,
+		TracingMaxNumAnnotations:              32,
+		TracingMaxNumMessageEvents:            128,
+		TracingMaxNumLinks:                    128,
+		MetadataURL:                           "http://169.254.169.254/computeMetadata",
+		IamURL:                                "https://iamcredentials.googleapis.com",
+		ServiceControlCredentials:             nil,
+		BackendAuthCredentials:                nil,
+		TranscodingAlwaysPrintPrimitiveFields: false,
```
Comment (JLXIA) [inline src/go/options/common.go]: These are not necessary.  In go, the default value for bool is false.

File: tests/integration_test/transcoding_print_options/transcoding_print_options_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Google LLC
```
Comment (JLXIA) [inline tests/integration_test/transcoding_print_options/transcoding_print_options_test.go]: what is the plan for re-organize these tests?

Maybe just put into grpc_test/grpc_integration_test.go

File: docker/generic/start_proxy.py [hunk 3]
```diff
@@ -489,6 +489,24 @@ def make_argparser():
     Default value: Not enabled. Please provide the certificate and key files
     /etc/nginx/ssl/backend.crt and /etc/nginx/ssl/backend.key.''')
 
+    # always_print_primitive_fields.
+    parser.add_argument('--transcoding_always_print_primitive_fields',
+                        action='store_true',
+                        help=argparse.SUPPRESS)
```
Comment (qiwzhang) [inline docker/generic/start_proxy.py]: My understanding is SUPRESS is to hide the flag.  I don't think we need to hide it

File: src/go/options/common.go (hunk 1 above)
Comment (qiwzhang) [reply src/go/options/common.go]: then, we should remove other fields with default values too., such as TracingProjectId,  and ServiceControlCredentials.

File: src/go/options/common.go [hunk 4]
```diff
@@ -49,6 +49,10 @@ type CommonOptions struct {
 	ServiceControlCredentials *IAMCredentialsOptions
 	// Configures the identity used when making requests to backends.
 	BackendAuthCredentials *IAMCredentialsOptions
+
```
Comment (nareddyt) [inline src/go/options/common.go]: These should go in config generator options, not common options. None of these flags are needed by the ADS Bootstrapper, they are only needed by the config generator.

Comment (nareddyt) [reply src/go/options/common.go]: Similar for the credentials above, but it's a little late to change that.

File: src/go/options/common.go (hunk 1 above)
Comment (TAOXUY) [reply src/go/options/common.go]: Just be consistent with other null flags

File: tests/integration_test/transcoding_print_options/transcoding_print_options_test.go (hunk 2 above)
Comment (TAOXUY) [reply tests/integration_test/transcoding_print_options/transcoding_print_options_test.go]: working on it.

moved to grpc_transcoding_test

File: src/go/options/common.go (hunk 1 above)
Comment (JLXIA) [reply src/go/options/common.go]: Try to clean up in seperate PR.

Comment (TAOXUY) [reply src/go/options/common.go]: Gocha.

**Pull request merged**